server/src/main/scala/org/http4s/server/middleware/Timeout.scala

```diff
@@ -21,9 +21,7 @@ object Timeout {
       http: Kleisli[F, A, Response[G]])(
       implicit F: Concurrent[F],
       T: Timer[F]): Kleisli[F, A, Response[G]] =
-    http
-      .mapF(respF => F.race(respF, T.sleep(timeout) *> timeoutResponse))
-      .map(_.merge)
+    http.mapF(Concurrent.timeoutTo(_, timeout, timeoutResponse))
```

Comment:
you should be able to use syntax for the `Concurrent` op, `_.timeoutTo(timeout, response)`

Reply:
Thought about that before the PR. IMHO, this is simple enough, and small enough, not to make it worth another `syntax` import.